keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -478,46 +487,84 @@ Because Windows privileged containers will work much differently than Linux priv
-  This functionality will most-likely not be ready during Kubernetes v1.23 and any changes made to how volume mounts work would be done before this features becomes stable.
-
-- Client libraries such as https://pkg.go.dev/k8s.io/client-go/rest#InClusterConfig may be updated to prefix paths with `$CONTAINER_SANDBOX_MOUNT_POINT` if the environment variable is set for Windows so these libraries will work in `hostProcess` containers.
-The decision to update client libraries (or not) will be postponed until the above mentioned merged container/OS filesystem investigations are concluded. Until then various workloads running in `hostProcess` containers need to communicate with the cluster can inject the `$CONTAINER_SANDBOX_MOUNT_POINT` environment variables into a kubeconfig file manually. Here is an example of how to do this today - https://github.com/jsturtevant/sig-windows-tools/blob/c9be1f0a9e95a34fda91bb7e8fc519e3447d8d93/hostprocess/calico/kube-proxy/start.ps1#L44-L52.
-  - Note: it is not possible to feature-gate this behavior in client libraries and because of this the functionality should not be added to client libraries after `hostProcess` containers while this feature is in `alpha`.
-  - [kubernetes/kubernetes#104490](https://github.com/kubernetes/kubernetes/pull/104490) adds support for `HostProcess` containers to the golang client library.
-- Named Pipe mounts will **not** be supported. Instead named pipes should be accessed via their path on the host (\\\\.\\pipe\\*).
-  - The following error will be returned if `hostProcess` containers attempt to use name pipe mounts - https://github.com/microsoft/hcsshim/blob/358f05d43d423310a265d006700ee81eb85725ed/internal/jobcontainers/mounts.go#L40.
-- Unix domain sockets mounts support will be added before `HostProcess` containers graduate to `stable`. For `alpha` and `beta` Unix domain sockets can be accessed via their paths on the host like named pipes.
-  - The Windows APIs needed to support mounting unix domain socket mounts in `hostProcess` containers was introduced in Windows Server Ver, 2004. Microsoft is planning on backporting these APIs to Windows Server 2019 (min support Windows Server OS) to provide a consistent user experience.
-- Mounting directories from the host OS into `hostProcess` containers will work just like with normal containers - kubelet will explicitly block this scenario.
-- All other volume types supported for normal containers on Windows will work with `hostProcess` containers.
+There will be two different behaviors for how volume mounts are configured in `hostProcess` containers.
+
+- **Bind Mounts**
+  - With the approach Window's bind-filter driver will be used to create a view that merges the host's OS filesystem with container-local volumes.
+  - When `hostProcess` containers are started a new volume will be created which contains the contents of the contaner image.
+    This volume will be mounted to `c:\hpc`.
```

Comment:
Shadowing already occurs for things like auto-mounted service account tokens, but we were careful to use kubernetes-specific paths (`/var/run/secrets/kubernetes.io/...`). Is `c:\hpc` sufficiently unique/namespaced to only apply to kubernetes?

Reply:
We had some discussions around this and do believe `c:\hpc` is sufficiently unique here.
@dcantah in case you have any comments.

Reply:
We look for an annotation in the runtime to configure the rootfs location as well. "microsoft.com/hostprocess-rootfs-location": "C:\some\unique\path"

Reply:
so from a container perspective, will things inside containers start hard-coding `c:\hpc` and relying on that as a well-known path?

Reply:
unless you want `c:\hpc` to become de-facto API, I'd recommend using a per-instance `c:\kubernetes\${random-uuid}` would prevent hard-coding dependencies on a particular path

Reply:
@brasmith-ms @dcantah @jsturtevant  - do we have any concerns with `c:\hpc` becoming the de-facto location?
The behavior of using a static path like `c:\hpc` aligns much more closely with the behaviors of regular containers on Windows which I think is desirable here.

Reply:
The reason I think we initially moved away from the random guid was because the bindmount changes this doesn't affect the host filesystem anymore and one of the challenges we had previously was that we needed the CONTAINER_SANDBOX_MOUNT_POINT env to find mount points.  Now that mounts are predictable this isn't an issue and it might be ok to be random.  

A few concerns I would have are:

- If I want the path added to the containers `PATH` env, how would I accomplish that? Does that path automatically get added? Can I override if I don't want it automatically?
- If I create a container that has binaries in folder off this path such as `c:\{random guid}\binaries\`  and would like it added to the `PATH` in the container, how would I do that?

This could be solved by CONTAINER_SANDBOX_MOUNT_POINT but sounds like it might not be desirable to have that long term?

Reply:
I'd prefer a static path, and we expose an annotation to be able to set it to something other than C:\hpc if needed. James touches on the initial pain point of a random path not known before launching a container which is how the heck do you actually reference it in a pod spec without that environment variable which I'd like to wind down use of..